src/ripple/basics/impl/Log.cpp

```diff
@@ -289,23 +289,41 @@ Logs::fromString (std::string const& s)
     return lsINVALID;
 }
 
-// Replace the first secret, if any, with asterisks
+// Attempt to prevent sensitive information from appearing in log files by
+// redacting it with asterisks.
 std::string
 Logs::scrub (std::string s)
 {
-    using namespace std;
-    char const* secretToken = "\"secret\"";
-    // Look for the first occurrence of "secret" in the string.
-    size_t startingPosition = s.find (secretToken);
-    if (startingPosition != string::npos)
+    auto scrubber = [&s](char const* token)
     {
-        // Found it, advance past the token.
-        startingPosition += strlen (secretToken);
-        // Replace the next 35 characters at most, without overwriting the end.
-        size_t endingPosition = std::min (startingPosition + 35, s.size () - 1);
-        for (size_t i = startingPosition; i < endingPosition; ++i)
-            s [i] = '*';
-    }
+        auto first = s.find(token);
+
+        // If we have found the specified token, then attempt to isolate the
+        // sensitive data (it's enclosed by double quotes, followed by a comma)
+        // and mask it:
+
+        if (first != std::string::npos)
+            first = s.find ('\"', std::min(first + std::strlen(token), s.size() - 1));
+
+        if (first != std::string::npos)
+        {
+            auto last = s.find(',', first);
+
+            if (last == std::string::npos)
+                last = s.size();
+
+            s.replace (first, last - first, last - first, '*');
+        }
+    };
+
+    scrubber ("\"seed\"");
+    scrubber ("\"seed_hex\"");
+    scrubber ("\"secret\"");
+    scrubber ("\"master_key\"");
+    scrubber ("\"master_seed\"");
+    scrubber ("\"master_seed_hex\"");
+    scrubber ("\"passphrase\"");
+
```

Comment:
As a minor exercise to increase our confidence this is a complete list, how about commenting the list of `JSS` in src/ripple/protocol/JsonFields.h with an attribute for each field that contains sensitive information.  For example:
```c++
JSS ( passphrase );                 // in [sensitive]: WalletPropose
```

As an example, the act of writing this comment caused me to wonder if the next `JSS` down from `passphrase` is also "sensitive" (`password`).